comit/src/expiries.rs

```diff
@@ -0,0 +1,488 @@
+//! Standardized COMIT Expiry Times
+//!
+//! The COMIT protocols rely on two expiry times in order for the swap to be
+//! atomic. One expiry for the alpha ledger and one for the beta ledger. The
+//! beta ledger expiry time must elapse before the alpha ledger expiry time.
+
+use crate::timestamp::{self, Timestamp};
+use time::Duration;
+
+// TODO: Currently we ignore deploy for Erc20.
+
+/// Current time as a UNIX timestamp as viewed by each of the underlying
+/// blockchains.
+pub trait CurrentTime {
+    fn alpha_current_time(&self) -> Timestamp;
+    fn beta_current_time(&self) -> Timestamp;
+}
+
+/// Data that defines and manipulates expiry times.
+///
+/// Language note: We use the term 'expiries' (plural of expiry) to make it
+/// explicit that there are two expiry times however we refer it as singular
+/// since it is the struct, or pair of expiries, that we are referring to.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct Expiries<C> {
+    /// Configuration values for calculating transition periods.
+    config: Config,
+    /// Blockchain connectors for getting the current time.
+    connector: C,
+    /// The alpha ledger expiry time.
+    alpha_expiry: Timestamp,
+    /// The beta ledger expiry time.
+    beta_expiry: Timestamp,
+}
+
+impl<C> Expiries<C> {
+    /// Create an expiries that will be useful for the duration of `ttl`.
+    pub fn create(ttl: Duration, protocol: Protocol, connector: C) -> Expiries<C>
+    where
+        C: CurrentTime,
```

Comment:
You are repeating yourself all over the place here. Moving this bound to the impl block should fix that.

Type variables that are defined on the actual type will usually also need their bounds to be defined there (as in `impl<C> Expiries<C> where C: ...`.

You can always have multiple `impl` blocks if the bounds differ per function.

Reply:
Awesome, thanks.